public/app/features/explore/utils/links.test.ts

```diff
@@ -16,470 +16,577 @@ import { initTemplateSrv } from '../../../../test/helpers/initTemplateSrv';
 import { ContextSrv, setContextSrv } from '../../../core/services/context_srv';
 import { setLinkSrv } from '../../panel/panellinks/link_srv';
 
-import { getFieldLinksForExplore } from './links';
-
-describe('getFieldLinksForExplore', () => {
-  beforeEach(() => {
-    setTemplateSrv(
-      initTemplateSrv('key', [
-        { type: 'custom', name: 'emptyVar', current: { value: null } },
-        { type: 'custom', name: 'num', current: { value: 1 } },
-        { type: 'custom', name: 'test', current: { value: 'foo' } },
-      ])
-    );
-  });
-
-  it('returns correct link model for external link', () => {
-    const { field, range } = setup({
-      title: 'external',
-      url: 'http://regionalhost',
-    });
-    const links = getFieldLinksForExplore({
-      field,
-      rowIndex: ROW_WITH_TEXT_VALUE.index,
-      splitOpenFn: jest.fn(),
-      range,
+import { getFieldLinksForExplore, dataLinkHasAllVariablesDefined } from './links';
+
+describe('explore links utils', () => {
```

Comment:
This is a complicated diff, but all I did was add an outer describe block so tests for `getFieldLinksForExplore` and `dataLinkHasAllVariablesDefined` can go under it